src/main.rs

```diff
@@ -99,12 +101,19 @@ enum DocItemOrLegacy {
 }
 
 /// Returns a rfc145 doc-comment if one is present
-pub fn retrieve_doc_comment(node: &SyntaxNode, shift_headings_by: Option<usize>) -> Option<String> {
+pub fn retrieve_doc_comment(
+    node: &SyntaxNode,
+    shift_headings_by: Option<usize>,
+    file: &Path,
+) -> Option<String> {
     let doc_comment = get_expr_docs(node);
 
-    doc_comment.map(|doc_comment| {
+    // Doc comments can import external file via "import" in frontmatter
+    let content = get_imported_content(file, doc_comment.as_ref()).or(doc_comment);
+
+    content.map(|inner| {
```

Comment:
```suggestion
    content.map(|inner| {
        // Doc comments can import external file via "import" in frontmatter
        let inner_new = get_imported_content(file, inner).or(inner);
```

This way you can avoid the `Option` in the function